### ISSUE TITLE
Backwards compatibility with tools that are unaware of go modules

### ISSUE THREAD

**Issue by felixge** [maruel/panicparse#57]
Hi there,

thanks for this project, it seems to work very well : ).

Quick Question: Would you be interested in making the library backwards compatible so that it could be imported by go projects that haven't upgraded to go modules yet? This would mainly involve creating a v2 directory inside of the repository and moving the v2 code into a v2 directory as recommended here: https://blog.golang.org/v2-go-modules#TOC_3.

---

What I mean is that the code below works fine as long as you have `go.mod` file in your project.

```go
package main

import (
	"fmt"
	"github.com/maruel/panicparse/v2/stack"
)

func main() {
	fmt.Printf("%#v\n", stack.Goroutine{})
}
```

However, if you don't have a `go.mod` file, you get an error like this:

```
main.go:6:2: cannot find package "github.com/maruel/panicparse/v2/stack" in any of:
	/usr/local/Cellar/go/1.15.6/libexec/src/github.com/maruel/panicparse/v2/stack (from $GOROOT)
	/Users/bob/go/src/github.com/maruel/panicparse/v2/stack (from $GOPATH)
```

If you're interested, I might be able to send you patch for this, but I figured I reach out first to see how you feel about it.

No worries if you decide you don't want this compatibility, your work on this is much appreciated either way : ).

-- Felix

**Comment by maruel** [comment]
Hi!

I did experiment with this idea last summer and spent countless hours to try to make it work. See my tangentially related rant at https://maruel.ca/post/panicparse-1.6.0/.

From what I found with my experiments, to make it work well I would have had to have a multi-module repository, with a v2/go.mod file. It would have led to more problems, as multi-modules repositories have **surprisingly bad ergonomics**. I can't justify the costly maintenance costs here. It's really bad, trust me. I experimented with another toy repository, see https://github.com/maruel/msgbus/tree/06da573e64ca238e4c1b985630698511a14af955 and gosh it was painful. In this example v1 should have been at the root to work non-modules, but the bad ergonomics problem remains. `go test ./...` silently doesn't do the expected thing, godoc is confused, etc. I'm not alone with this opinion, see https://twitter.com/peterbourgon/status/1192775076472672256 and https://twitter.com/davecheney/status/1340977171272069121.

That's why that for non go modules aware use case I went with the vendor/ solution. It's great because when in go modules mode, the vendor/ directory is silently ignored.

You don't see the vendor/ directory when you visit https://github.com/maruel/panicparse because v2 doesn't have it, it doesn't need it. But if you fetch the v1 branch, you'll see it, e.g. https://github.com/maruel/panicparse/tree/v1.6.1/vendor/github.com/maruel/panicparse/v2/stack

TL;DR: vendor v2 yourself as long as you stay in non-modules mode, you'll save yourself **much** trouble.

**Comment by felixge** [comment]
@maruel thank you for taking the time to type this thoughtful response. I've never tried to make a multi-module repo myself, but I've had my [own frustrations](https://twitter.com/felixge/status/1320485256605192192) with go modules, so I totally trust you when you say that things are very bad and complicated 🙃.

I've done the manual vendoring for now that you suggested.

Thanks again for working on this library and for the fast response.
